flink-libraries/flink-table/src/main/scala/org/apache/flink/table/functions/ScalarFunction.scala

```diff
@@ -58,6 +60,11 @@ abstract class ScalarFunction extends UserDefinedFunction {
 
   override def toString: String = getClass.getCanonicalName
 
+  final def functionIdentifier: String = {
```

Comment:
The  `UserDefinedFunction` has been declared as an abstract class in the newest master when we adding `open` and `close` default implementation. I think the `functionIdentifier` can be the same and will not hurt anything.